pkg/dns/serviceresolver.go

```diff
@@ -239,7 +244,7 @@ func (b *ServiceResolver) Records(dnsName string, exact bool) ([]msg.Service, er
 		glog.V(4).Infof("Answered %s:%t with %#v", dnsName, exact, services)
 		return services, nil
 	}
-	return nil, nil
+	return nil, errNoSuchName
```

Comment:
same question here, this is a recognized service with no endpoints, right? will returning errNoSuchName trigger a recursive lookup we don't want?